### PR TITLE
Add example for differential drive robot controller by ROS

### PR DESCRIPTION
**Pull request by kaaatsu32329**

### CODE REVIEW

File: arci-ros/src/msg.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@ ros_nalgebra::rosmsg_include!(
     sensor_msgs / JointState,
     sensor_msgs / Joy,
     std_msgs / Header,
+    std_msgs / Float64,
```
Comment (OTL) [inline arci-ros/src/msg.rs:19]: Please sort alphabetically.

```suggestion
    std_msgs / Float64,
    std_msgs / Header,
```

File: arci-ros/src/ros_control/joint_velocity_controller.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    publisher: rosrust::Publisher<Float64>,
+    _subscriber: rosrust::Subscriber,
+    motor_velocity_state: Arc<Mutex<f64>>,
+    is_reverse_direction: bool,
+}
+
+impl JointVelocityController {
+    pub fn new(pub_topic: &str, sub_topic: &str, reverse: bool, index: usize) -> Self {
+        let motor_velocity_state = Arc::new(Mutex::new(0.));
+        let aaa = motor_velocity_state.clone();
```
Comment (OTL) [inline arci-ros/src/ros_control/joint_velocity_controller.rs:23]: Avoid meaning-less words `aaa`.

File: arci-ros/src/ros_control/joint_velocity_controller.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#[derive(Clone)]
+struct JointVelocityControllerInner {
+    publisher: rosrust::Publisher<Float64>,
+    _subscriber: rosrust::Subscriber,
+    motor_velocity_state: Arc<Mutex<f64>>,
+    is_reverse_direction: bool,
+}
+
+impl JointVelocityController {
+    pub fn new(pub_topic: &str, sub_topic: &str, reverse: bool, index: usize) -> Self {
```
Comment (OTL) [inline arci-ros/src/ros_control/joint_velocity_controller.rs:21]: * pub_topic -> velocity_topic
* sub_topic -> joint_state_topic
* reverse -> is_reverse_direction
* index -> joint_index

File: arci-ros/src/ros_control/joint_velocity_controller.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+impl MotorDriveVelocity for JointVelocityController {
+    fn set_motor_velocity(&self, velocity: f64) -> Result<(), arci::Error> {
+        let vel = if self.0.is_reverse_direction {
+            -velocity
+        } else {
+            velocity
+        };
+
+        self.0.publisher.send(Float64 { data: vel }).unwrap();
```
Comment (OTL) [inline arci-ros/src/ros_control/joint_velocity_controller.rs:48]: Could you try to remove unwrap? (If it's difficult, don't mind, it's OK.)

File: openrr-base/examples/ros_control_diff_drive.rs [hunk 5]
```diff
@@ -0,0 +1,67 @@
+#[cfg(feature = "ros")]
```
Comment (OTL) [inline openrr-base/examples/ros_control_diff_drive.rs:1]: I feel better moving this into arci-ros/examples/.
What do you think?
I don't want to introduce ROS dependencies in openrr-base if possible.

Comment (kaaatsu32329) [reply openrr-base/examples/ros_control_diff_drive.rs:1]: I thought it was more meant to be an openrr-base example than an arci-ros example. But I agree that we don't want to put ROS dependencies in openrr-base and I strongly agree.

So I will move this code to arci-ros.

File: arci-ros/src/ros_control/joint_velocity_controller.rs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            velocity
+        };
+
+        match self.0.publisher.send(Float64 { data: vel }) {
+            Ok(_) => Ok(()),
+            Err(_) => Err(arci::Error::Connection {
+                message: "Error to publish motor velocity".to_owned(),
+            }),
+        }
+    }
```
Comment (OTL) [inline arci-ros/src/ros_control/joint_velocity_controller.rs:59]: I think

`.map_err()` is better to make it simple.

**Review by OTL — COMMENTED** [review]
LGTM! Let's got for merging this PR!

**Review by OTL — COMMENTED** [review]
LGTM! Let's got for merging this PR!

**Comment by OTL** [comment]
And we need to add apt-get install `libudev-dev` in the CI.

**Review by OTL — APPROVED** [review]
Thanks!

**Pull request merged**